fuel-merkle/src/sparse/merkle_tree.rs

```diff
@@ -569,6 +569,33 @@ mod test {
         assert_eq!(hex::encode(root), expected_root);
     }
 
+    #[test]
+    fn test_update_overwrite_key_2() {
```

Comment:
I'm not sure this is a useful test, since we already have tests to check:
- behaviour of overwriting keys
- update order independence
Do we really need this?

Reply:
Let's have combination of these two cases=D